src/json/json_schema.rs

```diff
@@ -328,4 +331,39 @@ mod tests {
 
         Ok(())
     }
+    #[test]
+    fn test_compare_enum() {
+        let mut en = BTreeSet::new();
+        en.insert("A".to_string());
+        en.insert("B".to_string());
+        let value = JsonSchema::Arr(Box::new(JsonSchema::Enum(en.clone())));
+        let schema = JsonSchema::Enum(en);
+        let name = "foo";
+        let result = schema.compare(&value, name);
+        assert_eq!(
+            result,
+            Valid::fail(format!("expected Enum got: {:?}", value)).trace(name)
+        );
+    }
+
+    #[test]
+    fn test_compare_enum_value() {
+        let mut en = BTreeSet::new();
+        en.insert("A".to_string());
+        en.insert("B".to_string());
+
+        let mut en1 = BTreeSet::new();
+        en1.insert("A".to_string());
+        en1.insert("B".to_string());
+        en1.insert("C".to_string());
+
+        let value = JsonSchema::Enum(en1.clone());
+        let schema = JsonSchema::Enum(en.clone());
+        let name = "foo";
+        let result = schema.compare(&value, name);
+        assert_eq!(
+            result,
+            Valid::fail(format!("expected {:?} but found {:?}", en, en1)).trace(name)
```

Comment:
```suggestion
            Valid::fail(format!("expected {:?} but found {:?}", en, en1)).trace(name)
```

Here too, insert the exact error.